File: frontend/src/features/report/services/reportService.js
import * as reportApi from "../api/reportApi";

export const reportService = {

    async getEquipmentReport() {
        const response = await reportApi.getEquipmentReport();
        return response.data;
    },

    async getIssueReport() {
        const response = await reportApi.getIssueReport();
        return response.data;
    },

    async getWorkOrderReport() {
        const response = await reportApi.getWorkOrderReport();
        return response.data;
    },

    async getEmployeeReport() {
        const response = await reportApi.getEmployeeReport();
        return response.data;
    }

};

File: backend/src/main/java/com/aayush/assetmanagement/report/service/ReportService.java
package com.aayush.assetmanagement.report.service;

import com.aayush.assetmanagement.report.dto.response.EmployeeReportResponse;
import com.aayush.assetmanagement.report.dto.response.EquipmentReportResponse;
import com.aayush.assetmanagement.report.dto.response.IssueReportResponse;
import com.aayush.assetmanagement.report.dto.response.WorkOrderReportResponse;

public interface ReportService {

    /**
     * Generates equipment report based on the logged-in user's role.
     *
     * ADMIN      -> All equipment
     * TECHNICIAN -> Equipment relevant to technician
     * EMPLOYEE   -> Equipment assigned to employee
     */
    EquipmentReportResponse getEquipmentReport();

    /**
     * Generates issue report based on the logged-in user's role.
     *
     * ADMIN      -> All issues
     * TECHNICIAN -> Assigned issues
     * EMPLOYEE   -> Reported issues
     */
    IssueReportResponse getIssueReport();

    /**
     * Generates work order report.
     *
     * Accessible to ADMIN and TECHNICIAN only.
     */
    WorkOrderReportResponse getWorkOrderReport();

    /**
     * Generates employee report.
     *
     * Accessible to ADMIN only.
     */
    EmployeeReportResponse getEmployeeReport();

}